File: app/src/main/java/com/androidacademy/msk/exerciseproject/di/module/IntroModule.java
package com.androidacademy.msk.exerciseproject.di.module;

import android.content.Context;
import android.support.annotation.NonNull;

import com.androidacademy.msk.exerciseproject.data.Storage;
import com.androidacademy.msk.exerciseproject.di.scope.ActivityScope;
import com.androidacademy.msk.exerciseproject.screen.intro.IntroPresenter;

import dagger.Module;
import dagger.Provides;

@Module
public class IntroModule {

    @Provides
    @ActivityScope
    @NonNull
    public IntroPresenter providePresenter(@NonNull Storage storage) {
        return new IntroPresenter(storage);
    }

    @Provides
    @ActivityScope
    @NonNull
    public Storage provideStorage(@NonNull Context context) {
        return new Storage(context);
    }

}

File: app/src/main/java/com/androidacademy/msk/exerciseproject/di/module/IntroModule.kt
package com.androidacademy.msk.exerciseproject.di.module

import com.androidacademy.msk.exerciseproject.data.network.shared_preferences.Storage
import com.androidacademy.msk.exerciseproject.screen.intro.IntroPresenter
import toothpick.config.Module


class IntroModule : Module() {
    init {
        bind(IntroPresenter::class.java).singletonInScope()
        bind(Storage::class.java).singletonInScope()
    }
}